[Blog] Aaditya's journey from Contributor to GSoC Intern @Layer5 (#2882)
layer5io/layer5: src/collections/blog/2022/2022-06-13-aaditya-gsoc-blogpost/index.mdx

---

> title: "My Journey as a Contributor to GSoC Intern @Layer5"

Reviewer: ```suggestion
title: "My Journey as a Contributor to GSoC Intern at Layer5"
```

subtitle: "Nothing but a magical experience"
date: 2022-06-13 08:00:00 -0630
author: Aaditya Narayan Subedy
thumbnail: ./aaditya-subedy-layer5-intern.png

> category: Community

Reviewer: ```suggestion
category: Programs
```

> description: "Aaditya Narayan Subedy's journey from a Contributor @Layer5 to getting an GSoC internship @Layer5"

Reviewer: ```suggestion
description: "Aaditya Narayan Subedy's journey from a Contributor at Layer5 to getting an GSoC internship @Layer5"
```

tags:
 - Community

>  - Projects

Reviewer: Theese tags are not appropriate. Use Programs, Experience, Gsoc etc

 - News
type: Blog
published: true
---

import { BlogWrapper } from "../../Blog.style.js";

<BlogWrapper>

### Early Interest in Coding

I grew up with an immense interest in computers. I used to be fascinated by how programs run on the computer that makes such an impact on the people using it. This always motivated me to work in the field of computers.

As I grew up I was introduced to a broad spectrum of coding languages that enabled us to make the software that we used on the computer. These languages piqued my interest even more, and I started to explore and make projects. The more I understood them the more interesting it got. Slowly I realized that I just don’t want to make projects that satisfy my needs. I want to build something that will be on the internet and will be used by people around the world. With this goal in my mind, I started exploring and hit a gemstone that is Open source. I would be working on real-world programming solutions that will be used by people all around the world. This helped me to get started and got me excited about contributing to open-source projects.

> ### First PR @ Layer5

Reviewer: ```suggestion
### First PR at Layer5
```

> I still remember the first issue I worked on for Layer5 was the auto labeler on a pull request. I saw a good first issue tag and still not knowing what the task was, I stated my interest to work on the issue. The issue required knowledge of GitHub Actions which at that time I didn’t even know there was such a thing. This is one amazing thing about open source, there is a community that is always there to help you. After doing some self-research and asking folks in the community I was finally able to solve the issue. I still remember the excitement when my PR got merged. I was enthralled to see that my code was actually working and being useful to the community. This motivated me to make more and more contributions.

Reviewer: You could add screenshots here. Maybe this one or of your fix
<img width="1122" alt="Screenshot 2022-06-14 at 3 51 31 PM" src="https://user-images.githubusercontent.com/85789734/173555624-69719627-d8e9-48f2-a90e-9499bc64c2ae.png">
.

> I still remember the first issue I worked on for Layer5 was the auto labeler on a pull request. I saw a good first issue tag and still not knowing what the task was, I stated my interest to work on the issue. The issue required knowledge of GitHub Actions which at that time I didn’t even know there was such a thing. This is one amazing thing about open source, there is a community that is always there to help you. After doing some self-research and asking folks in the community I was finally able to solve the issue. I still remember the excitement when my PR got merged. I was enthralled to see that my code was actually working and being useful to the community. This motivated me to make more and more contributions.

Reviewer: You can add a section for tips to make your way through the first contribution, even when you're doing something new. Breaking up the para and conveying all the steps you took in detail.

### A Journey Never to forget

> With this small issue, my journey on Layer5 started. I gradually started learning about the project and started making some impactful contributions. I started to engage in the community more prominently. With the help of these contributions, I learned so many new technologies, and techniques. Not to mention the community also helped me increase my confidence. The weekly meetings are responsible for this. The way the meeting required me to explain my fixes and give updates regarding the issues I was working on, helped me in getting vocal and comfortable with talking to various people.

Reviewer: Screenshot of a weekly meeting with you presenting maybe?

> With this small issue, my journey on Layer5 started. I gradually started learning about the project and started making some impactful contributions. I started to engage in the community more prominently. With the help of these contributions, I learned so many new technologies, and techniques. Not to mention the community also helped me increase my confidence. The weekly meetings are responsible for this. The way the meeting required me to explain my fixes and give updates regarding the issues I was working on, helped me in getting vocal and comfortable with talking to various people.

Reviewer: Talk more about these new technologies and techniques. Demonstrate this entire journey until you finally landed gsoc.

> I feel really excited that I have been accepted as a Google Summer of Code Intern for this Summer of 2022. I am looking forward to contributing more and making an impact here. I also want to state that this will not be the end. I will be engaged with this community as long as I can and will try to give back as much as I can.

Reviewer: Mention your plans/agenda items you wish to accomplish by the end of the internship.

---
title: "My Journey as a Contributor to GSoC Intern at Layer5"
subtitle: "Nothing but a magical experience"
date: 2022-06-13 08:00:00 -0630
author: Aaditya Narayan Subedy
thumbnail: ./aaditya-subedy-layer5-intern.png
category: Programs
description: "Aaditya Narayan Subedy's journey from a Contributor at Layer5 to getting an GSoC internship at Layer5"
tags:
 - Community
 - Projects
 - News
type: Blog
published: true
---
import { Link } from "gatsby";
import { BlogWrapper } from "../../Blog.style.js";
import aadityaFirstPr from "./aaditya-first-pr.png"
import aadityaPresenting from "./aaditya-presenting.png"

<BlogWrapper>

### Early Interest in Coding 

> I grew up with an immense interest in computers. I used to be fascinated by how programs run on the computer that makes such an impact on the people using it. This always motivated me to work in the field of computers. 

Reviewer: ```suggestion
I had a strong interest in computers growing up. I used to find it fascinating how computer programs operate because they have such a tremendous impact on their users. This has always motivated me to work in the field of computers.```

> As I grew up I was introduced to a broad spectrum of coding languages that enabled us to make the software that we used on the computer. These languages piqued my interest even more, and I started to explore and make projects. The more I understood them the more interesting it got. Slowly I realized that I just don’t want to make projects that satisfy my needs. I want to build something that will be on the internet and will be used by people around the world. With this goal in my mind, I started exploring and hit a gemstone that is Open source. I would be working on real-world programming solutions that will be used by people all around the world. This helped me to get started and got me excited about contributing to open-source projects. 

Reviewer: ```suggestion
As I grew up, I was introduced to a broad spectrum of coding languages that enabled us to make the software that we used on the computer. These languages piqued my interest even more. I started exploring and made projects. The more I understood them, the more interesting they got. Slowly I realized that I don’t want to make projects that satisfy just my needs. I want to build something that will be on the internet and will be used by people around the world. With this goal in my mind, I started exploring and hit a gemstone that is open source. I would be working on real-world programming solutions that will be used by people all around the world. This helped me to get started and got me excited about contributing to open source projects. 
```

### First PR at Layer5 

> I still remember the first issue I worked on for Layer5 was the auto labeler on a pull request. I saw a good first issue tag and still not knowing what the task was, I stated my interest to work on the issue. The issue required knowledge of GitHub Actions which at that time I didn’t even know there was such a thing. After getting the issue assigned, I started researching what GitHub Actions were. I found out that these actions helped to automate workflows that help in building, testing, and deploying software. After getting the basics down I looked for if there are any GitHub Actions already present for the auto labeling of the PR, and to my surprise, there were a couple of them. 

Reviewer: ```suggestion
I still remember the first issue I worked on for Layer5 was the auto labeler on a pull request. I spotted a good first issue tag and expressed interest in working on it despite not knowing what the task was. The issue required knowledge of GitHub Actions which at that time I didn’t even know there was such a thing. After getting the issue assigned, I started researching what GitHub Actions were. I found out that these actions helped automate workflows to build, test, and deploy software. After getting the basics down, I checked if there were any existing GitHub Actions for the auto labeling of the PR. To my surprise, there were a couple of them. 
```

> So before actually making a PR, I experimented with the Actions extensively in my private repo. I remember making about 20-25 PR in my private repo just to get the auto labeler working fine. After I verified I got the desired results, I made a PR. I also got to discuss my approach in the development call. I was so nervous at that time since that was the first meeting I attended where I had to make people understand what I did and what approaches I took. I was quite thrilled when my code got merged. I was enthralled to see that my code was working and being useful to the community. This motivated me to make more and more contributions. 

Reviewer: ```suggestion
So before actually making a PR, I experimented with the actions extensively in my private repo. I remember making about 20-25 PR in my private repo to get the auto labeler working fine. After verifying that I got the desired results, I made a PR. I also got to discuss my approach in the development call. I was so nervous at that time since that was the first meeting I attended where I had to make people understand what I did and the strategies I used. I was quite thrilled when my code got merged. I was enthralled to see that my code was working and being useful to the community. This motivated me to make more and more contributions. 
```

Reviewer: Link to Meshery Development Call to https://layer5.io/community/calendar

<img src={aadityaFirstPr} alt="Aaditya's first PR merged" />

### A Journey Never to forget 

> With this small issue, my journey on Layer5 started. I gradually started learning about the project and started making some impactful contributions. I started to engage in the community more prominently. With the help of these contributions, I learned new technologies, and techniques. Contributing to the codebase gave me so much exposure to React.js, that it would not have been possible if I did it on my own. I took up issues that used React.js and started solving them, and in no time I was comfortable with it. I am thankful, that I didn’t wait for me to learn more about react and then start contributing. I have always applied this principle, learning more by doing actual stuff rather than sitting there and getting stuck in the tutorial loop. 

Reviewer: ```suggestion
My Layer5 journey began with this small issue. I gradually started learning more about the projects and made some impactful contributions. I started to engage in the community more prominently. With the help of these contributions, I learned new technologies, and techniques. Contributing to the codebase gave me so much exposure to React.js, which would not have been possible if I had done it on my own. I took up issues that used React.js and started solving them, and in no time, I was comfortable with it. I am thankful that I didn’t wait for me to learn more about react and then start contributing. I have always applied this principle, learning more by doing actual stuff rather than sitting there and getting stuck in the tutorial loop. 
```

> Another new technology I learned was Jekyll, which is a static site generator. This too was a technology that I had not heard about. I started contributing to issues involving Jekyll and slowly understood the basics and how the laguage works. I was very happy that I was able to make the [compatibility matrix](https://docs.meshery.io/project/compatibility-matrix), using Jekyll. Now even my project for GSoC is written in Jekyll. So, from not knowing what Jekyll is, to making some big changes, did make me happy. 

Reviewer: ```suggestion
Another new technology I discovered was Jekyll, a static site generator. I was unfamiliar with this technology as well. I started contributing to issues involving Jekyll and slowly understood the basics and how the language works. I was very happy that I was able to make the [compatibility matrix](https://docs.meshery.io/project/compatibility-matrix), using Jekyll. Even my project for GSoC is written in Jekyll. So, going from not knowing what Jekyll is, to making some big changes, did make me happy. 
```

> The community also helped me increase my confidence. The weekly meetings are responsible for this. The way the meeting required me to explain my fixes and give updates regarding the issues I was working on, helped me in getting vocal and comfortable with talking to various people. 

Reviewer: ```suggestion
The community also helped me boost my confidence. The weekly meetings are responsible for this. The way the meetings required me to explain my fixes and give updates regarding the issues I was working on, helped me get vocal and comfortable with talking to various people. 
```

<img src={aadityaPresenting} alt="Aaditya Presenting" /> 

> I feel excited that I have been accepted as a Google Summer of Code Intern for this Summer of 2022. I am looking forward to contributing more and making an impact here. The project assigned to me is to make a dashboard at <a href="smp-spec.io/dashboard">SMP Website</a>. Here the results from the CNCF Cluster labs that run [here](https://github.com/layer5io/meshery-smp-action/actions) are to be shown graphically.  I also hope to learn more about Cloud Native technologies and make contributions on the backend too. I also want to state that this will not be the end. I will be engaged with this community as long as I can and will try to give back as much as I can. 

Reviewer: ```suggestion
I am excited to be accepted as a Google Summer of Code Intern for this Summer of 2022. I am looking forward to contributing more and making an impact here. The project assigned to me is to create a dashboard on <a href="smp-spec.io/dashboard">SMP Website</a>. Results from the [CNCF Cluster labs](https://github.com/layer5io/meshery-smp-action/actions) will be displayed here graphically.  I wish to learn more about cloud native technologies and make contributions on the backend too. I also want to state that this will not be the end. I will be engaged with this community as long as I can and will try to give back as much as I can. 
```

### Appreciation for the Community 

> I finally would like to thank the Layer5 Community for being such an amazing help when I would get stuck on some issue. I learned one very important thing here, there is always something you can learn and there is no harm in asking stuff if you don’t know. It’s kinda hard to find such a welcoming community, there was not a day that I felt completely lost here and that is what I adore about this community. 

Reviewer: ```suggestion
Finally, I would like to thank the Layer5 Community for always being so supportive whenever I needed it. I took away a very significant lesson from this experience, there is always something to learn, and there is no harm in asking questions if you don't know the answer. It’s kinda hard to find such a welcoming community. There was not a day that I felt completely lost here, and that is what I adore about this community.
```

> Keep Meshing Thank you 

Reviewer: ```suggestion
Keep Meshing. Thank you.
```
